ethereumj-core/src/main/java/org/ethereum/mine/BlockMiner.java

```diff
@@ -265,7 +266,13 @@ protected void restartMining() {
                 MinerIfc localMiner = config.getBlockchainConfig()
                         .getConfigForBlock(miningBlock.getNumber())
                         .getMineAlgorithm(config);
-                localMiner.setListeners(listeners);
+                if (localMiner instanceof EthashMiner) {
+                    Set<EthashMinerListener> ethashMinerListeners = listeners.stream()
```

Comment:
Maybe it would be better to do filtering in `EthashMiner.setListeners()` to keep the main flow clean?